.github/workflows/build.yaml

```
@@ -220,6 +220,35 @@ jobs:
           name: ${{ matrix.config.sys}}-${{ matrix.config.env }}-${{ matrix.config.compiler }}-testdir.tar.xz
           path: testdir.tar.xz
 
+  build_macos_universal:
+    name: MacOS Universal Binary
+    runs-on: macos-12
+    env:
+      CMAKE_GENERATOR: Ninja
+    steps:
+      - name: Get source
+        uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: |
+          HOMEBREW_NO_AUTO_UPDATE=1 HOMEBREW_NO_INSTALL_CLEANUP=1 \
+              brew install ninja
+      - name: Setup env
+        run: |
+          if [ "${{ matrix.config.compiler }}" = "gcc" ]; then
+            echo "CC=gcc" >> $GITHUB_ENV
+            echo "CXX=g++" >> $GITHUB_ENV
+          else
+            echo "CC=clang" >> $GITHUB_ENV
+            echo "CXX=clang++" >> $GITHUB_ENV
+          fi
+      - name: Build universal binary
+        run: ci/build-universal
+      - name: Archive universal binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: universal-binary
```

I think it would be a good idea to use a more specific name. Maybe

```suggestion
          name: macOS binary
```

?